test/cpp/auto_parallel/spmd_rule_test.cc

```diff
@@ -230,6 +230,84 @@ TEST(MatmulSPMDRule, Ctor) {
   check_partial_dims(inferred_dist_attrs.second[0], {0});
   VLOG(4) << "test11 done." << std::endl << std::endl << std::endl;
 }
+
+TEST(IndexPut, Ctor) {
+  // build input data class
+  std::vector<int64_t> x_shape = {64, 64, 64};
+  std::vector<int64_t> indice_shape = {32};
+  std::vector<int64_t> value_shape = {32};
```

Comment:
in this case, if `value_shape=[32, 64]`, can run normally on single device, but can run normally on multi device? especially dims_mapping of x is {-1, 0, 1}
tips: After using this spmd rule, the local shape will be: `x_shape=[64, 64, 32], indice_shape=([32], [32]), value_shape=[32, 64]`, Can `index_put` operator will run successfully?